odc/stats/plugins/tcw_percentiles.py

```diff
@@ -76,10 +110,15 @@ def native_transform(self, xx):
         """
         Loads data in its native projection.
         """
-        bad = enum_to_bool(
-            xx["fmask"], ("nodata", "cloud", "shadow")
-        )  # a pixel is bad if any of the cloud, shadow, or no-data value
-        bad |= xx["nbart_contiguity"] == 0  # or the nbart contiguity bit is 0
+        mask = xx["fmask"]
+        nodata = enum_to_bool(mask, ("nodata",))
+        non_contiguent = xx["nbart_contiguity"] == 0
+        bad = nodata | non_contiguent
+        # Now exclude cloud and cloud shadow (including buffered pixels)
+        for cloud_class, c_filter in self.cloud_filters.items():
+            cloud_mask = enum_to_bool(mask, (cloud_class,))
+            cloud_mask_buffered = mask_cleanup(cloud_mask, mask_filters=c_filter)
+            bad = cloud_mask_buffered | bad
```

Comment:
Can we keep the fall-back behaviour here where there is no dilation? for the sake of sanity check/test and backward compatibility.

Reply:
Added support for no dilation and added a log message so if done unintentionally we can see it in the log.